build.gradle

```diff
@@ -8,7 +8,7 @@ plugins {
 }
 
 group = 'com.gradle'
-version = layout.projectDirectory.file('release/version.txt').asFile.text.trim()
+version = version()
```

Comment:
Extract the call to version() into a variable above, like you did for the release notes provider. I would call it `pluginVersion` to avoid any naming collision/confusion.

Reply:
Done - but just so I know, what is the issue with referencing `project.version` elsewhere in the script?